pkg/apiserver/handlers.go

```diff
@@ -534,3 +534,28 @@ func (r *APIRequestInfoResolver) GetAPIRequestInfo(req *http.Request) (APIReques
 
 	return requestInfo, nil
 }
+
+// IsAPIRequest looks for patterns like /prefix/version/resource and /prefix/group/version/resource. It if matches, returns true.
```

Comment:
> Why not also do the extraction and return these things? I don't like having extraction and predicate code possibly evolving separately.

They aren't all required.  The other realistic solution is alter https://github.com/kubernetes/kubernetes/blob/master/pkg/apiserver/handlers.go#L442 to return a structured error type here: https://github.com/kubernetes/kubernetes/blob/master/pkg/apiserver/handlers.go#L450 and https://github.com/kubernetes/kubernetes/blob/master/pkg/apiserver/handlers.go#L462, but not here https://github.com/kubernetes/kubernetes/blob/master/pkg/apiserver/handlers.go#L475.

I'm not a fan of introspecting errors to go down expected code paths, but it could be done.